### DISCUSSION TITLE
Problem: when register one client with two contrats with two publish Apis on two diferents gateways who configured with differents index on elasticseach database

### DISCUSSION THREAD

**Discussion by GFlo69** [apiman/apiman#1485] [Help]
Apiman version : 2.1.0
Elasticseach : 7.x

Case :
Apigateway 1 with index prefix api1index [OK]
Apigateway 2 with index prefix api2index [OK]
Api A publish on apigateway 1 [OK]
Api B publish on  apigateway 2 [OK]
Create contrat beetween Client App Z on Api A [OK]
Register client Z [OK]
Create contrat beetween Client App Z on Api B [OK]
Register client App Z [KO]

Explain : when register client, apimanager send to apigateway all contract rattach to client. In my case, api A is not reconized on Apigateway 2 and api B is not reconized on Apigateways 1.

In this case, i'm in obligation to publish Api A ans api B on two gateways 1 and 2

**Comment by msavy** [comment]
If you want multiple gateways to share the same configuration (i.e. recognise same APIs), they need need to be looking at the *same* persist storage data -- the same registry configuration i.e. same index names.

e.g:

Start vert.x gateway A and vert.x gateway B. Both are configured to use the *same* Elasticsearch index for the registry section of the configuration.

Add only one gateway to the API Manager, the other gateway will pick up the same configuration because it's looking at the same persistent store.

If you use different index names, they will be independent and you would get the behaviour you are describing (which is deliberate and covers a different use-case, for example production environment vs deployment).

Make sense?
